### ISSUE TITLE
Feature Request - Standalone backups

### ISSUE THREAD

**Issue by keithf4** [pgbackrest/pgbackrest#1053]
Didn't see any open issues for this and not sure if anything like this is on the roadmap.

Requesting to allow running backups without the requirement of a dedicated repository as the destination. All options necessary can either be passed directly on the command line or provided via a file. Of course this limits the options available, so not looking to have the standalones be as feature complete as a repository backup (Ex. no automatic retention management).

Normally the archive_command would need a dedicated repository as well to run. However, allowing the archive_command to work in a standalone manner as well could allow this. It would just need a target location to place the WAL files (and any other files necessary to ensure proper archiving). Could even allow multiple target locations.

For reference where I'm coming from with this, I used to previously use https://github.com/omniti-labs/omnipitr which was designed entirely around only doing standalone backups. Made it very easy to just spit out another backup at a different location whenever it was needed for any reason (dev/testing/etc).

**Comment by dwsteele** [comment]
No, there's nothing like this in the backlog right now. I think configuration could be a bit tricky unless we just used the `repo-*` options somehow -- and I don't think we could do that because WAL would continue to need to be sent to the regular repo or there would be a gap. If the ask is to copy to a local directory that would be easier, but I don't think it would be a very good feature if you couldn't copy to the available storage types.

**Comment by keithf4** [comment]
So I can see how it's tricky with the current system. And even with the omnipitr setup, you would need the archive_command to be going to a different location to support a one off backup to a separate destination.

Some thoughts on this (checked project page and didn't see anything like this mentioned)

- Add the ability to extract a standalone backup from the repo. Could maybe use the existing restore mechanism to just give a destination and it would automatically tie all the files required for that backup into a single file (tar, zip, etc). I get people could likely do something like this themselves, but having a convenience command available for pulling out a specific, standalone backup to be able to move around as needed could be nice. It could also include all the checksum info as well for reference.

- Add the ability to have different archive destinations outside the repository for a single stanza. Then it requires no changes other than a pgbackrest config change and the next time the archive_command runs, it just sends it to multiple locations instead. Could possibly even have a flag for it whether you would want failure to that secondary location to cause an error to be returned to postgres.

Not sure how realistic/sensible these are. Just throwing out some thoughts

**Comment by sfrost** [comment]
Hey Keith,

Some of what you're talking about is actually being actively worked on, though we are calling it 'multi-repo' support, where you can configure pgbackrest to push WAL to multiple destinations, and then following from that we intend to support the ability to possibly push backups from one repo to another as you mention.  Would be great to hear if the multi-repo support that we're working on might be able to address the use-cases you're contemplating here, and if not, how we might address those use-cases with multi-repo support.

While having a way to get a single backup+WAL files out as a tar file does sound interesting, how would you imagine tablespaces working with that?  Maybe if we add support for tar in general (which we are contemplating, but for different reasons) then it might make this reasonably straight-forward to add but I don't think we could really have a *single* tar file due to having to deal with tablespaces, at which point I'm not sure if it's really that much better than just doing a regular restore and then doing whatever you want with the results of that..?  If you could help explain what the use-case is for this, that would be helpful.
